prime-router/src/main/kotlin/common/HttpClientUtils.kt

```diff
@@ -496,78 +517,47 @@ class HttpClientUtils {
             httpClient: HttpClient? = null,
         ): HttpResponse {
             return runBlocking {
-                (httpClient ?: createDefaultHttpClient(accessToken)).request(url) {
-                    this.method = method
-                    timeout {
-                        requestTimeoutMillis = timeout
-                    }
-                    url {
-                        queryParameters?.forEach {
-                            parameter(it.key, it.value.toString())
+                // use{...} will automatically close the client
+                // https://ktor.io/docs/client-create-and-configure.html#close-client
+                (httpClient ?: getDefaultHttpClient()).use { client ->
+                    client.request(url) {
+                        this.method = method
+                        timeout { requestTimeoutMillis = timeout }
+                        url {
+                            queryParameters?.forEach {
+                                parameter(it.key, it.value)
+                            }
                         }
-                    }
-
-                    headers?.let {
-                        headers {
-                            headers.forEach {
-                                append(it.key, it.value)
+                        headers?.let {
+                            headers {
+                                headers.forEach { append(it.key, it.value) }
                             }
                         }
-                    }
-                    acceptedContent?.let {
-                        accept(acceptedContent)
-                        contentType(acceptedContent)
-                    }
-                    jsonPayload?.let {
-                        setBody(jsonPayload)
+                        accessToken?.let {
+                            headers {
+                                append("Authorization", "Bearer $accessToken")
+                            }
+                        }
+                        acceptedContent?.let {
+                            accept(acceptedContent)
+                            contentType(acceptedContent)
+                        }
+                        jsonPayload?.let {
+                            setBody(jsonPayload)
+                        }
                     }
                 }
             }
         }
 
         /**
-         * Create a http client with sensible default settings
+         * Get a http client with sensible default settings
```

Comment:
This comment block no longer works imo since we turned this into a simple getter now and create the client elsewhere.

Reply:
This looks good to me now, I either commented on the wrong thing or commented on an old revision maybe.